packages/mui-base/src/Portal/Portal.test.tsx

```diff
@@ -79,25 +79,25 @@ describe('<Portal />', () => {
         </Portal>
       </div>,
     );
-    const rootElement = document.querySelector('#test1');
-    expect(rootElement.contains(document.querySelector('.woofPortal1'))).to.equal(true);
-    expect(rootElement.contains(document.querySelector('.woofPortal2'))).to.equal(false);
+    const rootElement = document.querySelector<HTMLDivElement>('#test1');
+    expect((rootElement as any).contains(document.querySelector('.woofPortal1'))).to.equal(true);
+    expect((rootElement as any).contains(document.querySelector('.woofPortal2'))).to.equal(false);
   });
 
   it('should unmount when parent unmounts', () => {
     function Child() {
-      const containerRef = React.useRef();
+      const containerRef = React.useRef<HTMLDivElement>();
       return (
         <div>
-          <div ref={containerRef} />
-          <Portal container={() => containerRef.current}>
+          <div ref={containerRef as any} />
```

Comment:
You can remove the `as any` here (same on other occurences)

Reply:
If I fail to mention the type I see a "null" unassignable or "undefined" unassignable error so I changed any to specific types.

Reply:
You should only use `as any` as a last resort.
Here, I think that doing `const containerRef = React.useRef<HTMLDivElement>(null);` fixes the issue (never truly understood why though)